Kashoo/synctos
# Issue 331: Stub the requireAdmin function

# Description

Adds a stub for the `requireAdmin` function so that document definitions that call it from, for example, a custom action can be verified with a test fixture.

# Testing

Follow these instructions for Sync Gateway [2.1.0](https://packages.couchbase.com/releases/couchbase-sync-gateway/2.1.0/couchbase-sync-gateway-enterprise_2.1.0_x86_64.tar.gz):

1. Generate a sync function from `test-environment-auth-doc-definitions.js`:

```bash
./make-sync-function -j test/resources/test-environment-auth-doc-definitions.js build/test-sync-function.txt
```

2. Copy and paste the generated sync function into a Sync Gateway configuration file.
3. Attempt to create a valid `requireAdminDoc` document via the public REST API (should be rejected with 403 Forbidden):

```
PUT /test/foobar HTTP/1.1
Host: localhost:4984
Content-Type: application/json
Authorization: Basic ********

{
	"type": "requireAdminDoc",
	"text": "foo"
}
```

4. Create a valid document via the admin REST API (should be accepted with 201 Created):

```
PUT /test/foobar HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "requireAdminDoc",
	"text": "foo"
}
```

# Related Issue

* GitHub issue link: #331
